google-cloud-spanner/src/test/java/com/google/cloud/spanner/TransactionChannelHintTest.java

```diff
@@ -0,0 +1,335 @@
+                    // Verify that the compressor name header is set.
+                    assertEquals(
+                        "gzip",
+                        headers.get(
+                            Metadata.Key.of(
+                                "x-response-encoding", Metadata.ASCII_STRING_MARSHALLER)));
+                    Attributes attributes = call.getAttributes();
+                    @SuppressWarnings({"unchecked", "deprecation"})
+                    Attributes.Key<InetSocketAddress> key =
+                        (Attributes.Key<InetSocketAddress>)
+                            attributes.keys().stream()
+                                .filter(k -> k.equals(TRANSPORT_ATTR_REMOTE_ADDR))
+                                .findFirst()
+                                .orElse(null);
+                    if (key != null) {
+                      if (call.getMethodDescriptor()
+                          .equals(SpannerGrpc.getExecuteStreamingSqlMethod())) {
+                        executeSqlLocalIps.add(attributes.get(key));
+                      }
+                      if (call.getMethodDescriptor().equals(SpannerGrpc.getStreamingReadMethod())) {
+                        streamingReadLocalIps.add(attributes.get(key));
+                      }
+                    }
+                    return Contexts.interceptCall(Context.current(), call, headers, next);
+                  }
+                })
+            .build()
+            .start();
+  }
+
+  @AfterClass
+  public static void stopServer() throws InterruptedException {
+    server.shutdown();
+    server.awaitTermination();
+  }
+
+  @BeforeClass
+  public static void disableLogging() {
+    Logger logger = Logger.getLogger("");
+    originalLogLevel = logger.getLevel();
+    logger.setLevel(Level.OFF);
+  }
+
+  @AfterClass
+  public static void resetLogging() {
+    Logger logger = Logger.getLogger("");
+    logger.setLevel(originalLogLevel);
+  }
+
+  @After
+  public void reset() {
+    mockSpanner.reset();
+    executeSqlLocalIps.clear();
+    streamingReadLocalIps.clear();
+  }
+
+  private SpannerOptions createSpannerOptions() {
+    String endpoint = address.getHostString() + ":" + server.getPort();
+    return SpannerOptions.newBuilder()
+        .setProjectId("[PROJECT]")
+        .setChannelConfigurator(
+            input -> {
+              input.usePlaintext();
+              return input;
+            })
+        .setCompressorName("gzip")
+        .setHost("http://" + endpoint)
+        .setCredentials(NoCredentials.getInstance())
+        .build();
+  }
+
+  @Test
+  public void testSingleUse_usesSingleChannel() {
+    try (Spanner spanner = createSpannerOptions().getService()) {
+      DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
+      try (ResultSet resultSet = client.singleUse().executeQuery(SELECT1)) {
+        while (resultSet.next()) {}
+      }
+    }
+    assertEquals(1, executeSqlLocalIps.size());
```

Comment:
A single-use read-only transaction only executes 1 RPC, so it would in any case just use 1 channel. Could we instead extend this test to execute more than 1 single-use read-only transaction and verify that each new single-use read-only transaction uses a new channel? (Assuming that the latter is what we expect, but I think it is, as we want round-robin channel pooling in this case)

Reply:
The same also applies to the other single-use tests in this file.

Reply:
Yeah, I think I included them as an afterthought. Can't recall why. Have removed the same.